server/src/Application/Services/Flashcard/FlashcardService.cs

```diff
@@ -42,6 +46,7 @@ public IReadOnlyCollection<Flashcard> GetFlashcardsBySetId(int flashcardSetId)
 
         flashcard.Question = dto.Question ?? flashcard.Question;
         flashcard.Answer = dto.Answer ?? flashcard.Answer;
+        _updatedFlashcardValidator.ValidateAndThrow(flashcard);
```

Comment:
At this point the information must be valid; validation should be at the top of the method (not sure why it is in controller, while in other cases it's in service, should be in service).